### PR TITLE
Space Exploration Project - Rockets, Missions and Profile pages

### PR DESCRIPTION
**Pull request by Ellon-M**
## Changes

 - Set up the project with `create-react-app` and linters for JavaScript and CSS.
 - Set up `redux toolkit` and configured the redux store.
 - Wrote action creators and actions for the rockets.
 - Fetched rocket data from the SpaceX API.
 -  Wrote action creators and actions for the missions.
 - Fetched mission data from the SpaceX API.
 - Rendered the rockets and the missions on the UI's of their respective pages.
 - Wrote conditional switch statements for the buttons to change the functionality when the state is altered.
 - Displayed the reserved rockets and joined missions on the `Profile.js` page.
 - Wrote unit tests for the rockets, missions, and navigation bar components.

### CODE REVIEW

File: src/components/Profile.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  return (
+    <>
+      <div className="profile">
+        <section className="section-rockets">
+          <h3 className="profile-heading">Reserved Rockets</h3>
+          <Table hover size="xl">
+            <tbody>
+              {rockets && rockets.map((rocket) => (rocket.reserved ? <tr key={rocket.id} className="profile-rows">{rocket.name}</tr> : <></>))}
+            </tbody>
```
Comment (danmainah) [inline src/components/Profile.js:21]: - [ ] Kindly ensure the reserved rockets are displayed using the filter method as you did in the missions, this is a project requirement 
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Ellon-M and @quavo19 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/Rocket.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    <li className={styles.rocket}>
+      <div className={styles.rocketImg}>
+        <img src={images[1]} alt={`SpaceX ${type}: ${name}`} />
+      </div>
+      <div className={styles.rocketDetails}>
+        <h4 className={styles.rocketName}>{name}</h4>
+        <p className={styles.rocketDesc}>
+          {reserved ? <span className={styles.reservedBanner}>reserved</span> : <></>}
+          {description}
+        </p>
```
[the remark this answers is not in the document]
Comment (quavo19) [reply src/components/Rocket.js:30]: @danmainah please do you mean the reserved badge?

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      "no-shadow": "off",
+      "no-param-reassign": 0,
+      "max-len": ["error", { "code": 160 }],
+      "import/prefer-default-export": "off"
+    },
+    "ignorePatterns": [
+      "dist/",
+      "build/"
+    ]
+  }
```
Comment (emmiiorji) [inline .eslintrc.json:31]: Could you make sure you use the unmodified given linters configuration for react/redux projects? The [.eslintrc.json](https://github.com/microverseinc/linters-config/blob/master/react-redux/.eslintrc.json) and [linters.yml](https://github.com/microverseinc/linters-config/blob/master/react-redux/.github/workflows/linters.yml) files is supposed to match yours.

Linters represent coding standards you share with your team and prevent divergent coding styles. Most of the time, they also enforce the best coding practices.

Your .eslintrc.json vs correct config:
image

Your linters.yml vs correct config:
image

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @Ellon-M and @quavo19, 

# Status: Changes Required ♻️

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 A little bit more

## Good Highlights✔️

- Good commit messages
- Adhered to Gitflow
- Requested changes have been implemented


 ## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by levy002 — APPROVED** [review]
## Hello @Ellon-M  & @quavo19 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Status: APPROVED :fire:  :fire: 

## Highlights
- Professional README file :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by quavo19** [comment]
@levy002 thank you sir

**Pull request merged**